Fix for job tasks to be sorted by task kay alphabetically

goda opened the pull request:
Fix for #286 - adding a method in `Job` class that sorts its `tasks` by `task_key` whenever the class is serialized. This will ensure that when no job task's have been changed that terraform doesn't think an update is needed.

osoucy commented on laktory/models/resources/databricks/job.py:
```diff
@@ -862,3 +862,8 @@ def terraform_properties(self) -> dict:
             d[k] = _clusters
 
         return d
+
```
Can you move this method after the declaration of the job attributes and before the `# Resource Properties` block? This would be line 758.

osoucy commented:
Can you update the `CHANGELOG.md` file to include in the updated section:
`* `Job` to automatically alphabetically sorts `tasks` [[#286](https://github.com/okube-ai/laktory/issues/277)]`

osoucy commented:
Can you update the `test_job.py` file by changing the order of the  tasks in the `job` definition ? This will validate that the new sorting works as expected. Final job definition could be something like:
```

job = Job(
    name="job-stock-prices",
    clusters=[
        {
            "name": "main",
            "spark_version": "14.0.x-scala2.12",
            "node_type_id": "Standard_DS3_v2",
        },
    ],
    tasks=[
        {
            "depends_ons": [{"task_key": "ingestion"}],
            "pipeline_task": {"pipeline_id": "${resources.pl-spark-dlt.id}"},
            "task_key": "pipeline",
        },
        {
            "task_key": "view",
            "sql_task": {
                "query": {"query_id": "456"},
                "warehouse_id": "123",
            },
        },
        {
            "job_cluster_key": "main",
            "notebook_task": {
                "notebook_path": "job/ingest_stock_prices",
            },
            "task_key": "ingestion",
        },
    ],
    email_notifications={
        "on_duration_warning_threshold_exceededs": ["info@okube.ai"],
        "on_failures": ["info@okube.ai"],
        "on_starts": ["info@okube.ai"],
        "on_successes": ["info@okube.ai"],
    },
)
```

goda commented:
Pushed new commit with the requested changes:
- CHANGELONG.md updated to reflect the change
- Changed order of tasks in test_job `job` object to ensure the sorting of the tasks works
- Moved the `sort_tasks` method right after the attribute declarations in `Job` class
